src/test/java/giis/eshopcontainers/e2e/functional/tests/WebAggAPITests.java

```diff
@@ -0,0 +1,142 @@
+package giis.eshopcontainers.e2e.functional.tests;
+
+import com.google.gson.Gson;
+import giis.eshopcontainers.e2e.functional.common.BaseAPIClass;
+import giis.eshopcontainers.e2e.functional.model.Order;
+import giis.eshopcontainers.e2e.functional.model.OrderItem;
+import giis.retorch.annotations.AccessMode;
+import giis.retorch.annotations.Resource;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.ResponseHandler;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.BasicResponseHandler;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.List;
+
+public class WebAggAPITests extends BaseAPIClass {
+
+    // Test to ensure items are added to the basket correctly
+    @Resource(resID = "identity-api", replaceable = {})
+    @AccessMode(resID = "identity-api", concurrency = 50, sharing = true, accessMode = "READONLY")
+    @Resource(resID = "basket-api", replaceable = {})
+    @AccessMode(resID = "basket-api", concurrency = 30,sharing = true, accessMode = "READWRITE")
+    @Resource(resID = "eshopUser", replaceable = {})
+    @AccessMode(resID = "eshopUser", concurrency = 1, accessMode = "READWRITE")
+    @Test
+    @DisplayName("testAddProductsBasketWebAgg")
+    void testAddProductsBasket() throws IOException, InterruptedException {
+        // Initialize Gson for JSON serialization/deserialization
+        Gson gson = new Gson();
+
+        addItemsToBasket();
```

Comment:
What is this? the test data setup? the test execution? Here is where a comment would be useful